src/syntax/rules.rkt

```diff
@@ -32,35 +32,80 @@
       (match-define (list rules groups types) ruleset)
       (list (filter rule-ops-supported? rules) groups types)))))
 
-(define (type-of-rule input output)
+(define (type-of-rule input output [prec 'binary64]) ; see below
+  (define type
+    (cond
+      [(list? input) (operator-info (car input) 'otype)]
+      [(list? output) (operator-info (car output) 'otype)]
+      [else
+        (error 'type-of-rule "Could not compute type of rule ~a -> ~a"
+                input output)]))
+  (if (equal? type 'real) prec type)) ; 'if' can still emit real. TODO: fix
```

Comment:
Not sure what this comment means. Worrying!

Reply:
Oh, right, this. This is one of the few cases where `if` is treated as an operator and it's return 'type' is `real` since it is unparameterized. One possible fix is to put a case in `operator-info` for `if` statements that returns the `repr-of` the `ift` branch.

Reply:
Hmm, `operator-info` is supposed to mostly be a direct hash table lookup. What about putting a branch here instead of in `operator-info`?

Reply:
Ah, that's better.